src/mirth/elab.mth

```diff
@@ -299,6 +301,14 @@ def(elab-arrow-step-word!, Arrow Word -- Arrow +Elab,
     elab-token@ token-args-0 # high order not yet supported
     elab-word-sig! elab-arrow-apply-sig-type!)
 
+def(elab-arrow-step-stack-assertion!, Arrow Token -- Arrow +Elab,
+    token-brace-arg
+    dip(elab-ctx@)
+    elab-type-stack! drop nip
+    dup morphism-type-new! TMorphism
+    elab-arrow-apply-sig-type!
```

Comment:
this felt like cheating :)

Reply:
This is really good!

Reply:
Oh actually I just realised this probably doesn't work . . .

Because IIRC `elab-arrow-apply-sig-type!` will freshen the morphism type before doing matching, so it's going to interact weirdly with type variables in the signature.

Probably we are better off doing some sort of manual matching here, not sure.